retrievalmarket/impl/provider.go

```diff
@@ -388,48 +395,61 @@ func (p *Provider) HandleQueryStream(stream rmnet.RetrievalQueryStream) {
 	sendResp(answer)
 }
 
-func (p *Provider) getPieceInfoFromCid(ctx context.Context, payloadCID, pieceCID cid.Cid) (piecestore.PieceInfo, bool, error) {
-	cidInfo, err := p.pieceStore.GetCIDInfo(payloadCID)
+// Given the CID of a block, find a piece that contains that block.
+// If the client has specified which piece they want, return that piece.
+// Otherwise prefer pieces that are already unsealed.
+func (p *Provider) getPieceInfoFromCid(ctx context.Context, payloadCID, clientPieceCID cid.Cid) (piecestore.PieceInfo, bool, error) {
+	// Get all pieces that contain the target block
+	piecesWithPayloadCID, err := p.piecesWithBlock(payloadCID)
```

Comment:
nit: variable name should be just `pieceCid`.

Reply:
Before the function had a parameter called pieceCID, so I didn't want to also use pieceCID as the variable name for the cids of pieces containing the payload CID.
I think it's clearer if the two variables are named clientPieceCID and piecesWithPayloadCID

Reply:
Changed the name of piecesWithPayloadCID to piecesWithTargetBlock. Kind of a mouthful but I hope that's clearer :)